src/microns_to_nwb/tools/ophys/ophys.py

```diff
@@ -0,0 +1,113 @@
+from phase3 import nda, func
+from pynwb.image import GrayscaleImage
+from pynwb.ophys import (
+    RoiResponseSeries,
+    Fluorescence,
+    ImageSegmentation,
+    OpticalChannel,
+)
+
+from tools.nwb_helpers import check_module
+
+
+def add_summary_images(field_key, nwb):
+    ophys = nwb.create_processing_module("ophys", "processed 2p data")
+
+    correlation, average = (nda.SummaryImages & field_key).fetch1("correlation", "average")
+
+    correlation_image = GrayscaleImage(correlation)
+    average_image = GrayscaleImage(average)
+
+    ophys.add(correlation_image)
+    ophys.add(average_image)
+
+
+def add_plane_segmentation(field_key, nwb, imaging_plane, image_segmentation):
+    ps = image_segmentation.create_plane_segmentation(
+        name=f"PlaneSegmentation{field_key['field']}",
+        description="output from segmenting my favorite imaging plane",
+        imaging_plane=imaging_plane,
+    )
+    ps.add_column("mask_type", "type of ROI")
+
+    image_height, image_width = (nda.Field & field_key).fetch1("px_height", "px_width")
+
+    mask_pixels, mask_weights, mask_ids, mask_types = (nda.Segmentation * nda.MaskClassification & field_key).fetch(
+        "pixels", "weights", "mask_id", "mask_type", order_by="mask_id"
+    )
+
+    # Reshape masks
+    masks = func.reshape_masks(mask_pixels, mask_weights, image_height, image_width)
+
+    for image_mask, mask_id, mask_type in zip(masks, mask_ids, mask_types):
+        ps.add_roi(
+            image_mask=image_mask,
+            id=mask_id,
+            mask_type=mask_type,
+        )
+
+    return ps
+
+
+def add_roi_response_series(scan_key, field_key, nwb, plane_segmentation):
+    frame_times = (nda.FrameTimes & scan_key).fetch1("frame_times")
+
+    data = np.vstack((nda.Fluorescence() & field_key).fetch("trace", order_by="mask_id")).T
+
+    rt_region = plane_segmentation.create_roi_table_region(
+        region=list(range(data.shape[1])), description=f"all rois in field {field_key['field']}"
+    )
+
+    roi_response_series = RoiResponseSeries(
+        name=f"RioResponseSeries{field_key['field']}",
+        description=f"traces for field {field_key['field']}",
+        data=H5DataIO(data, compression=True),
+        rois=rt_region,
+        timestamps=H5DataIO(frame_times, compression=True),
+        units="n/a",
+    )
+
+    fluorescence = Fluorescence()
+    fluorescence.add(roi_response_series)
+
+
+def add_ophys(scan_key, nwb):
+    device = nwb.create_device(
+        name="Microscope",
+        description="two-photon random access mesoscope",
+    )
+    ophys = check_module(nwb, "ophys")
+    image_segmentation = ImageSegmentation()
+    ophys.add(image_segmentation)
+    all_field_data = (nda.Field & scan_key).fetch(as_dict=True)
+    for field_data in all_field_data:
+        optical_channel = OpticalChannel(
+            name="OpticalChannel",
+            description="an optical channel",
+            emission_lambda=500.0,
+        )
```

Comment:
Is this a new OpticalChannel being made every time? Or is there one OpticalChannel that links to each separate plane?

Reply:
Yeah I just checked in the example file you sent and each one has a different object ID. It's pretty subtle and the difference isn't likely to affect much, but I'd still try to only link to one if possible

Reply:
OpticalChannels aren't linked to, they are contained within the ImagingPlane, so you must create a unique one each time. I'm not crazy about this, but that's the current state of the schema for better or worse

Reply:
Hmm OK then